stuff2sort/System.py

```diff
@@ -0,0 +1,1150 @@
+from Generators._helper import _flatten, get_linkatoms_ang
+
+#   // TODOS & NOTES //
+#   TODO:
+#   - Add the option to read atom input from a list (if that's what we want, wait for Florians opinion)
+#   - Add logger
+#   NOTE:
+
+#   // CLASS & METHOD DEFINITIONS //
+class SystemInfo():
+
+    '''
+    This Class Reads And Stores Information About The System
+    '''
+
+    def __init__(self, input_dict) -> None:
+
+        '''
+        ------------------------------ \\
+        EFFECT: \\
+        --------------- \\
+        Calls Functions To Read Information About The System\\
+        ------------------------------ \\
+        INPUT: \\
+        --------------- \\
+        input_dict: dict -> Dictionary Of The User Parameter Input \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        NONE \\
+        ------------------------------ \\
+        '''
+
+        self.input_dict = input_dict
+
+        #   Make a list of all topology files
+        self.topology_list = self.getincludelist(self.input_dict['topologyfile'])
+
+        # Read The Different Types Of Molecules In The System
+        self.list_of_molecules = self.read_molecules()
+
+        # Read The Charges For All Atoms
+        self.charge_vector = []
+        for element in self.list_of_molecules:
+            self.charge_vector.extend(self.readcharges(element))
+
+        #   Read All Atom Lists
+        #   XX AJ currently I'm assuming we're always having separate files for atom indices, only remove this comment when we finally decided so or add the possibility for lists
+        self.qmatomslist = self.read_atoms_list(self.input_dict['qmatomslist'])
+        self.activeatomslist = self.read_atoms_list(self.input_dict['activeatomslist'])
+        if input_dict['useinnerouter']:
+            self.inneratomslist = self.read_atoms_list(self.input_dict['inneratomslist'])
+            self.outeratomslist = self.read_atoms_list(self.input_dict['outeratomslist'])
+        #   If we're not using inner/outer we're keeping the lists empty (currently necessary for some functions but that might change later)
+        else:
+            self.inneratomslist = []
+            self.outeratomslist = []
+
+        #   Read Connectivity
+        self.connectivity_list = self.read_connectity_from_topology(self.input_dict['topologyfile'])
+
+        #   Read Initial Geometry
+        #   XX AJ I would prefer only one function here independent of the file type and only make that distinction within the function. I'll get back to that later when I'm writing GeneratorGeometries.py
+        if self.input_dict['coordinatefile'][-4:] == ".gro":
+            #logger(logfile, "Reading geometry (.gro)...\n")
+            # XX AJ if we rewrite gro files to g96 files, the following function can be deleted and we can read the g96 file with geometry.readg96 afterwards
+            self.initial_geometry = geometry.readgeo(self.input_dict['coordinatefile'])
```

Comment:
self.geometry_initial

Reply:
Is self.list_geometry_initial better? Because it has also the data type